sbi/neural_nets/flow.py

```diff
@@ -458,75 +736,82 @@ def build_zuko_maf(
             sample is, for example, a time series or an image.
         z_score_y: Whether to z-score ys passing into the network, same options as
             z_score_x.
-        hidden_features: Number of hidden features.
-        num_transforms: Number of transforms.
-        embedding_net: Optional embedding network for y.
-        residual: whether to use residual blocks in the coupling layer.
-        randperm: Whether features are randomly permuted between transformations or not.
-        kwargs: Additional arguments that are passed by the build function but are not
-            relevant for maf and are therefore ignored.
+        hidden_features: The number of hidden features in the flow. Defaults to 50.
+        num_transforms: The number of transformations in the flow. Defaults to 5.
+        embedding_net: The embedding network to use. Defaults to nn.Identity().
+        **kwargs: Additional keyword arguments to pass to the flow constructor.
 
     Returns:
-        Neural network.
+        ZukoFlow: The constructed Zuko normalizing flow model.
     """
-    x_numel = batch_x[0].numel()
-    # Infer the output dimensionality of the embedding_net by making a forward pass.
-    check_data_device(batch_x, batch_y)
-    check_embedding_net_device(embedding_net=embedding_net, datum=batch_y)
-    embedding_net.eval()
-    y_numel = embedding_net(batch_y[:1]).numel()
-    if x_numel == 1:
-        warn(
-            "In one-dimensional output space, this flow is limited to Gaussians",
-            stacklevel=1,
-        )
+
+    x_numel, y_numel = get_numel(batch_x, batch_y, embedding_net)
 
     if isinstance(hidden_features, int):
         hidden_features = [hidden_features] * num_transforms
 
-    if x_numel == 1:
-        maf = zuko.flows.MAF(
-            features=x_numel,
-            context=y_numel,
-            hidden_features=hidden_features,
-            transforms=num_transforms,
+    build_nf = getattr(zuko.flows, which_nf)
+
+    if which_nf == "CNF":
```

Comment:
so continuous flows only have one transform, so I get why this if case is necessary. Can you add some comments here to make this a bit less opaque?